### PR TITLE
first pass at declarative install

### PR DESCRIPTION
**Pull request by Gregory-Pereira**
/cc @sabre1041 
Changes:
- swapping clientserver Openshift route to an ingress
- conditionally deploy clientserver manifests
- add TAS pull-secret to the charts (either by providing a value or will default back to the cluster pull-secret)
- remove hard-coded manifest creation commands from tas installer script

### CODE REVIEW

**Review by sabre1041 — CHANGES_REQUESTED** [review]
A few initial changes requested

File: charts/trusted-artifact-signer/values.yaml [hunk 1]
```diff
@@ -5,8 +5,15 @@ global:
...
...eir@redhat.com"},"registry.redhat.io":{"auth":"fHVoYy1wb29sLTU1ODc1NmZkLTNjZTctNGUzMC1hMTE5LTc1NmEyZjEwZDRkZTpleUpoYkdjaU9pSlNVelV4TWlKOS5leUp6ZFdJaU9pSXdOREpqTXpBMVl6azVOVEkwWXpOaU9HRXhPV0ZoT1RBMk9EaGtNbVprT0NKOS5maXBCYXcyOXVpdU5WTk9sVndfa0lhWlRndkdJU3J3TWhJVUpackNCblM1UTBSRmlPbFRzTlVab0Nob1g3SFVyekVkX1ZUWHRnT1FoZ1QxR1FkWFNrOWQ5dzk5akZDd3pvOE5FVzYyMWZmZTh1RHVLR3c4NFpjTm1kMlo3ZDl3OHJGZXBFa3dUeTBuVzlnMnoyWEtnOGJhYmZaSUJoR244U24tM2g1MUxWdEhBbTVSbEN6T1ZpdGdhdk9YdUd2ZHdibU5GdUFZODJ3d3NqZWVkZDVFQVlFTm5UZHk3ZlNaWmdIY0FYN1l3UkNNRXVnNjNVVzR4TGZPdEhtTkJrWDY5SXc3bFY5WklUcjNBY3ZUSGswZ0JWSmswVGJHSHFWeko4S2hYTWRSVDRmMHNYZUpGWm5sMHMzbUJTajNtTUxSSHF6UHZzeVBxT3o2YWpubXUxTnJBeEFhb1F1VXloT1pyMjd6cVQ4SjlnTzNRVllDTXdHQ0xpeGlhRzhEd1R2cTF3eGl4dnRzcWVENkhVck1EemtSYnVQYnh0V3hPTmpZNnJDWVlCV3BxQUdlNlg0SnZEbXB0U3Y2ZjZ5THBieDRfLVBDN0Z0TzB1Nk9MSFlRTUpaSUhEYXRkOXJuakNhb3JkVzJxTldZTm5mczZKbHotMy1taGFWSG1BSjBYcmpVdGQ5UFBFREhZQndrb29XZUdIYjExU2xfY2RhdklPdU02NHBickkwR1dRXzJ2LWxCWmd4UzVzNFRoWURrMkUyOHRTd1ZZR3A0ckx5dXdGVHphSnQ5TVcxcms2ZW5ndGhQbEt6cnphTTBMWG51TWVyUmgyRXlaNkVyN1drUnZTN2p4U1lRVDQxM0lUdmNtcUZoM0UtSFdBXzB4SzZ2eUNyaVpLMGVxRWotT1RpSQ==","email":"grpereir@redhat.com"}},"orgId":"11009103","userId":"54342237"}
```
Comment (osmman) [inline charts/trusted-artifact-signer/values.yaml:10]: I think that this should be empty. You probably by mistake commit your pull secret credentials.

File: charts/trusted-artifact-signer/templates/tas-pull-secret.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{{ if .Values.configs.pull_secret.secret_create }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: pull-secret
+  namespace: {{ .Values.configs.pull_secret.namespace }}
+type: Opaque
+data:
+  {{- if .Values.configs.pull_secret.value }}
```
[the remark this answers is not in the document]
Comment (Gregory-Pereira) [reply charts/trusted-artifact-signer/templates/tas-pull-secret.yaml:9]: I tried it messing with it for 10 minutes and opted to just go with this, but if you can get it to work with ternary feel free.

File: examples/values-sigstore-openshift.yaml [hunk 3]
```diff
@@ -45,4 +45,10 @@ scaffold:
       http:
         hosts:
           - host: tuf.$OPENSHIFT_APPS_SUBDOMAIN
-            path: /
\ No newline at end of file
+            path: /
+
+configs:
```
Comment (sabre1041) [inline examples/values-sigstore-openshift.yaml:50]: `configs` is already declared above and this should be placed underneath it

File: examples/values-sigstore-openshift.yaml [hunk 4]
```diff
@@ -45,4 +45,10 @@ scaffold:
...
+  pull_secret:
+    secret_create: false
```
Comment (sabre1041) [inline examples/values-sigstore-openshift.yaml:52]: with `secret_create` being set to false, the pull secret is never actually being created.  When I set it to be `true`, the following is created:

```yaml
apiVersion: v1
kind: Secret
metadata:
  name: pull-secret
  namespace: sigstore-monitoring
type: Opaque
data:
  pull-secret.json: map[]
```

**Review by sabre1041 — CHANGES_REQUESTED** [review]
A few issues found

File: examples/values-sigstore-openshift.yaml (hunk 4 above)
Comment (Gregory-Pereira) [reply examples/values-sigstore-openshift.yaml:52]: this should be adressed from switching the data type to a string which has been done.

**Review by sabre1041 — APPROVED** [review]

**Pull request merged**